src/core/Akka.Streams/Implementation/ActorMaterializerImpl.cs

```diff
@@ -384,7 +384,13 @@ public override TMat Materialize<TMat>(IGraph<ClosedShape, TMat> runnable, Func<
         /// </summary>
         /// <param name="logSource">The source that produces the log events.</param>
         /// <returns>The newly created logging adapter.</returns>
-        public override ILoggingAdapter MakeLogger(object logSource) => Logging.GetLogger(System, logSource);
+        public override ILoggingAdapter MakeLogger(object logSource)
+        {
+            if (logSource is not LogSource s) return Logging.GetLogger(System, logSource);
```

Comment:
Use the old code if someone's not using a `LogSource` - I'm open to changing this so everything does it the new way.

Reply:
Done, went ahead and made this change.